tests/cases/integration/Repository/repository.persistence.phpt

```diff
@@ -96,6 +96,56 @@ class RepositoryPersistenceTest extends TestCase
 	}
 
 
+	public function testCountAfterRemoveAndFlushAndCount()
+	{
+		$author = new Author();
+		$author->name = 'The Imp';
+		$author->web = 'localhost';
+		$author->born = '2000-01-01 12:12:12';
+
+		$this->orm->authors->attach($author);
+
+		$publisher = new Publisher();
+		$publisher->name = 'Valyria';
+
+		$book = new Book();
+		$book->author = $author;
+		$book->title = 'The Wall';
+		$book->publisher = $publisher;
+		$book->translator = $author;
+
+		$book2 = new Book();
+		$book2->author = $author;
+		$book2->title = 'The Wall II';
+		$book2->publisher = $publisher;
+		$book2->previousPart = $book;
+
+		$this->orm->authors->persistAndFlush($author);
+
+		$authorId = $author->id;
+		$publisherId = $publisher->id;
+
+		$author = $this->orm->authors->getById($authorId);
+		$publisher = $this->orm->publishers->getById($publisherId);
+
+		foreach ($author->books as $book) {
+			$this->orm->books->remove($book);
+		}
+		$this->orm->books->flush();
+
+		$book3 = new Book();
+		$book3->author = $author;
+		$book3->title = 'The Wall III';
+		$book3->publisher = $publisher;
+
+		Assert::same(\count($author->books), 1); // I might expect this to be 0 before persisting but whatever
```

Comment:
First, you changed the $expected with $actual, then the error messages in the travis are probably wron.g

Second, if I a get it correctly, you expect this to be 1, but the code (count) returns a 0.
Is this the issue? Everything else just twist my head more, so I'm basically looking only on this code for now.

Reply:
Sorry, I confuse the order of arguments of `Assert::same` every time. But even then, the result should not change.

I expect this to be `0` because the new book was not persisted yet but `1` would be acceptable too, if we want to add it to `$author->books` immediately after `$book->author` is set.

The issue is the second assert, which does not make sense by any standard.

Reply:
> The actual issue is the second assert, which does not make sense by any standard.

Oookay, that's the problem, since the test fails before it and therefore it was not obvious.

> I expect this to be 0 because the new book was not persisted yet 

This should be 1 and it being it 0 is a bug. The count should return actual relationship state, not "what is persisted".

Reply:
>> The actual issue is the second assert, which does not make sense by any standard.
>
> Oookay, that's the problem, since the test fails before it and therefore it was not obvious.

Wait, why the second assert should be something different and what is should be?

Reply:
> Oookay, that's the problem, since the test fails before it and therefore it was not obvious.

Travis does fail in the second assert (note the missing comment next to the line in the test failure output):

https://travis-ci.org/nextras/orm/jobs/655337123#L311

> This should be 1 and it being it 0 is a bug. The count should return actual relationship state, not "what is persisted".

This assertion passes, which is correct according to the relationship state interpretation.

> Wait, why the second assert should be something different and what is should be?

Both assertions should pass (the count being 1 in both places).